net/net_tcp.c

```diff
@@ -852,6 +857,8 @@ struct tcp_connection* tcp_conn_create(int sock, union sockaddr_union* su,
 	if (!send2main)
 		return c;
 
+	c->send_sock = si;
```

Comment:
I don't think there's any need of a new `c->send_sock` in the connection structure, as you already have that information in the `c->rcv.bind_address` field.

Reply:
Fixed